src/overlay/SurveyDataManager.h

```diff
@@ -0,0 +1,206 @@
+#pragma once
+
+// Copyright 2024 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "main/Config.h"
+#include "overlay/Peer.h"
+#include "util/NonCopyable.h"
+#include "util/Timer.h"
+#include "util/UnorderedMap.h"
+
+#include "medida/meter.h"
+#include "xdr/Stellar-overlay.h"
+#include "xdr/Stellar-types.h"
+
+#include <chrono>
+#include <mutex>
+#include <optional>
+
+namespace stellar
+{
+
+enum class SurveyPhase
+{
+    // Survey is currently recording data
+    RECORDING,
```

Comment:
Nit: any chance we could either rename "recording" to collecting or rename "reporting" to complete/ready/something along those lines? Recording and reporting are perfectly accurate terms for the survey, it's just they're so similar sounding/looking words, that it makes reviewing harder.

Reply:
Ah, I can see how that's confusing at a glance. I changed "recording" to "collecting" in 8e51348e7f07bb7cfd01a63608dd9d6079769d3f

Reply:
thanks!